### ISSUE TITLE
Show TV/Radio group in the channel listing description in PVR

### ISSUE THREAD

**Issue by dagwieers** [add-ons/service.iptv.manager#76]
Since we now have many add-ons adding channels to PVR IPTV Simple ( and in some cases the same channel) it becomes important for users to know which group (add-on) this channel is part of.

![screenshot013](https://user-images.githubusercontent.com/388198/106530908-82d5ab80-64ed-11eb-868a-abd52ff7fb15.png)

cc @phunkyfish

**Comment by phunkyfish** [comment]
I don’t see the issue. They just need to select the group they are interested in instead of all channels.

**Comment by dagwieers** [comment]
It is not so much an issue, than it is an inconvenience. Selecting a group just to find if a channel is part of that group is counter intuitive and may not actually help that much. We have add-ons that expose 50 or more channels (TV Vlaanderen, Yelo).

Also, not every household member knows about channel groups, and this would require you to select all groups again after you stopped watching the channel you were looking for.

Being able to see in Kodi which group the selected channel belongs to is a lot more convenient.

**Comment by phunkyfish** [comment]
A channel can belong to multiple groups or a single group. I’m not sure how to depict what you want in that view.

**Comment by michaelarnauts** [comment]
I think we should solve this issue another way. For a user, it should not matter what group a channel is in. Duplicate channels should be hidden or merged somehow. Maybe something more in the lines of #43

**Comment by dagwieers** [comment]
@michaelarnauts The purpose is not just to differentiate between channels, but to know what group they are in. (For IPTV Manager it would identify the add-on it relates to)

#43 is a different issue.

**Comment by dagwieers** [comment]
> A channel can belong to multiple groups or a single group. I’m not sure how to depict what you want in that view.

The easiest option is to have the group names listed in the right pane.

| **Groups:** Catch-up TV & More, Belgium

**Comment by michaelarnauts** [comment]
> > A channel can belong to multiple groups or a single group. I’m not sure how to depict what you want in that view.
> 
> The easiest option is to have the group names listed in the right pane.
> 
> | **Groups:** Catch-up TV & More, Belgium

This would be an issue for Kodi then. I don't think there is a way to pass a description with a Channel.

**Comment by dagwieers** [comment]
It _is_ an issue for Kodi. This is a tracking ticket.

**Comment by phunkyfish** [comment]
Why is an issue for kodi? A channel can belong to any number of groups. It’s not feasible to show a list with possibly many items in the right pane.

**Comment by dagwieers** [comment]
> Why is this an issue for kodi?

It is not an issue, it is an enhancement.

> A channel can belong to any number of groups. It’s not feasible to show a list with possibly many items in the right pane. Nor is it correct to do so.

Most (if not all) channels are not part of many groups. But in the case there are many, we can do exactly the same thing as the Kodi category (they are cut off when they go beyond the foreseen box). At least in the majority of cases it would help understand which group(s) a channel belongs to. I think it would be extremely useful to our Kodi users.

> The UI is structured so the group is at the top, channel list on the left and channel details on the right.

Perfect, this is a channel detail, so it belongs on the right pane as I suggested.

PS When we discussed this before, one option was to have it added to the EPG description. For this we would not require a change to Kodi,. But decided that having it in Kodi would be visually more attractive (and skins could opt to use it differently).

**Comment by phunkyfish** [comment]
You are thinking of the UI in the wrong way. The parent of the entire screen is the channel group. What you are asking for is to display the other channel groups this channel belong too but only for all channels?

The skin doesn’t know this is the all channels group. It just display the channels listed without any knowledge of any other groups.

In short, what you are asking for is not an easy problem to solve in a sensible way.

**Comment by dagwieers** [comment]
I appreciate your input (that is why I added you in cc), but we are just discussing an idea here. Which starts with what is desirable.

I understand that the current code does not offer this information readily available. And that Skins do not have access to this information already. (Honestly, if they did that would be fantastic and changed the course of this process).

But what you are doing here is dismissing an idea on the basis that it would require changes to Kodi and skins, not on its own merits. That is the world upside down IMO. The code assists in what we think is needed for the UI, not the other way around.

Making the list of groups available as a variable to the skin, per channel/id, is in itself not a hard problem to solve. Putting this information on the right pane in a skin neither. But that is not even the purpose of this tracking ticket.  Its purpose is to have an entry point for discussing a perceived inconvenience (something we can point people to), discussing possible options openly and eventually improve Kodi in this direction.

We have worked like this before for all our add-on projects, and it has lead over time to various improvements in Kodi, ISA, ISH and Up Next.

**Comment by phunkyfish** [comment]
Fair point. This has pivoted how I think about that channels UI now. Because now many thousands of channels are now a distinct reality whereas before it was something that you would try in earnest to avoid.

**Comment by dagwieers** [comment]
@phunkyfish I am sure how we think about this will change when this gets used more widely. I would also like to see what @matthuisman's view is on this.

**Comment by phunkyfish** [comment]
As a workaround how does it look if the provider add-on name is pretended/appended to the channel name?

Far from ideal but might be an acceptable as a starting point.

**Comment by dagwieers** [comment]
That was one of the options indeed, however some add-on names are rather long (even the add-on id MSB).

Would be nice if we had superscript of subscript BB-codes for this ;-)

**Comment by matthuisman** [comment]
iptv merge by default adds the add-on name to the channels groups.
So users can go "Groups > Add-on Name" to view the channels for that add-on.
That can be changed / turned off.

If you did above, in the add-ons themselves you could have a shortcut to view the channels for that add-on
xbmc.executebuiltin('ActivateWindow(TVChannels,pvr://channels/tv/addon_name)')
(above is kodi 19 only - https://kodi.wiki/view/Opening_Windows_and_Dialogs)

Also, would it just be a skin change to show a channels groups somewhere? Maybe below the description?
Or does kodi not provide that information to skinners?

**Comment by phunkyfish** [comment]
Currently that information is not provided to skinners. Although I think we might be trying to make a concept fit that doesn’t. Groups can mean many things. For instance I have several groups which split the same source of channels in genres. If a user has another PVR add-on and iptvsimple it can get very messy with groups quickly.

Conceptually would it make more sense for a channel to have a provider field possibly?

**Comment by matthuisman** [comment]
Provider makes a lot of sense to me.
eg. TvHeadend, MythTV, Addon A, Addon B..
Leave groups for categories.

If it's just a new property, then we can use #KODIPROP in the playlists
and no new code would be required for IPTV Simple as that already parses #KODIPROP.

**Comment by phunkyfish** [comment]
In IPTV Simple it’s easy. But it would require an API change in kodi which means targeting kodi 20.

**Comment by dagwieers** [comment]
@matthuisman If the provider would be an add-on id, the implementation could use the add-on logo or add-on name as they see fit.

**Comment by matthuisman** [comment]
@dagwieers thats a good idea.
I wasn't sure if skinners have methods to get add-on info (name / logo) from just the addon id?

**Comment by dagwieers** [comment]
> @dagwieers thats a good idea.
> I wasn't sure if skinners have methods to get add-on info (name / logo) from just the addon id?

If not, they could be expanded by Kodi so skinners have them.

**Comment by phunkyfish** [comment]
I will look at adding support for providers in kodi 20. A provider can have a name, a type (thinking satellite, cable, aerial and iptv) and an icon. Anything else that would be needed?

**Comment by michaelarnauts** [comment]
> I will look at adding support for providers in kodi 20. A provider can have a name, a type (thinking satellite, cable, aerial and iptv) and an icon. Anything else that would be needed?

Sounds nice, maybe a Country? Although this might be more something that should be specified on a Channel.

**Comment by phunkyfish** [comment]
I thought of country but then for providers that span countries it’s not so useful. Language possibly.
